frappe/desk/doctype/dashboard_chart/dashboard_chart.py

```diff
@@ -173,6 +176,31 @@ def get_chart_config(chart, filters, timespan, timegrain, from_date, to_date):
 
 	return chart_config
 
+def get_heatmap_chart_config(chart, filters):
+	aggregate_function = get_aggregate_function(chart.chart_type)
+	value_field = chart.value_based_on or '1'
+	doctype = chart.document_type
+	datefield = chart.based_on
+	filters.append([doctype, datefield, '>', 'subdate(curdate(), interval 1 year)', False])
+
+	data = dict(frappe.db.get_all(
+		doctype,
+		fields = [
+			'unix_timestamp(date({datefield}))'.format(datefield=datefield),
```

Comment:
This will break for Postgres. [Ref](https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-EXTRACT)